ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1604,12 +1604,14 @@ private InstallSnapshotReplyProto notifyStateMachineToInstallSnapshot(
                     LOG.debug("{}: StateMachine could not install snapshot as it is not available", this);
                   }
                 }
-              // wait until the snapshot is installed successfully or exceptionally
-              }).join();
-        } catch (Throwable t) {
-          inProgressInstallSnapshotRequest.compareAndSet(firstAvailableLogTermIndex, null);
-          throw t;
-        }
+              // wait for 10 seconds for statemachine to install snapshot
+              }).get(10, TimeUnit.SECONDS);
```

Comment:
10 seconds is too long to block the main thread and wait for this executor thread to finish. I leader could assume this follower is not responding and stop the follower log appended from its side (is there a timeout?). Even if there is no timeout, I dont think we should block the main thread for more than a second.

Reply:
sure, let us set this to 1 second